Lib/unittest/test/testmock/testmock.py

```diff
@@ -1802,5 +1802,18 @@ def test_parent_attribute_of_call(self):
         self.assertEqual(type(call.parent().parent), _Call)
 
 
+    def test_parent_propagation_with_create_autospec(self):
+
+        def foo(a, b):
+            pass
+
+        mock = Mock()
+        mock.child = create_autospec(foo)
+        mock.child(1, 2)
+
+        self.assertRaises(TypeError, mock.child, 1)
+        self.assertEqual(mock.mock_calls, [call.child(1, 2)])
```

Comment:
What was the outcome here before your fix?

Reply:
`mock.mock_calls` would be an empty list though the `child` mock was called. It's a case where `create_autospec` returns a function object with mock attached to it in the `mock` attribute instead of a mock.